CONTRIBUTING.md

```diff
@@ -28,82 +28,103 @@ If any of this sounds interesting, I hope you consider getting involved with Ast
 ```
 
-Then bump the versions:
+Which will fail if the performance has regressed by **10%** or more.
+
+To update the times cd into the `packages/astro` folder and run the following:
 
 ```shell
-yarn changeset version
+node test/benchmark/build.bench.js --save
+node test/benchmark/dev.bench.js --save
 ```
 
-This should change package.json bumping version numbers and update changelogs. Inspect these to make sure they are what you expect.
+Which will update the build and dev benchmarks.
 
-Commit and push these changes, then run an npm publish for each of the packages that have changed.
 
-> **Important**! Ordering publishes can be important. If `@astrojs/parser` changes you should publish that before `astro`, for example.
+# Releasing Astro
 
-```shell
-cd packages/astro
-npm publish
-```
+_Note: Only priviledged contributors (L3+) can release new versions of Astro._
 
-## Running benchmarks
+The repo is set up with automatic releases, using the changeset GitHub action & bot. 
 
-We have benchmarks to keep performance under control. You can run these by running (from the project root):
+To release a new version of Astro, find the `Version Packages` PR, read it over, and merge it.
 
-```shell
-yarn workspace astro run benchmark
-```
+## Releasing PR preview snapshots
 
-Which will fail if the performance has regressed by **10%** or more.
+Our release tool `changeset` has a feature for releasing "snapshot" releases from a PR or custom branch. These are npm package publishes that live temporarily, so that you can give users a way to test a PR before merging. This can be a great way to get early user feedback while still in the PR review process.
 
-To update the times cd into the `packages/astro` folder and run the following:
+To release a snapshot, run the following locally:
 
 ```shell
-node test/benchmark/build.bench.js --save
-node test/benchmark/dev.bench.js --save
+# Note: XXX should be a keyword to identify this release. Ex: `--snapshot routing` & `--tag next--routing`
+
+# 1:
+yarn changeset version --snapshot XXX
+# 2: (Manual) review the diff, and make sure that you're not releasing more than you need to.
+git checkout -- examples/ docs/ www/
+# 3:
+yarn release --tag next--XXX
 ```
 
-Which will update the build and dev benchmarks.
+Full documentation: https://github.com/atlassian/changesets/blob/main/docs/snapshot-releases.md
+
+
+
+## Releasing in "next"/"pre" mode
+
+Sometimes, the repo enters "next" mode, which means that `main` is no longer releasing to `npm install astro` but is instead releasing to `npm install astro@next`. We do this from time-to-time to test large features before sharing them with the larger Astro audience.
+
+When in "next" mode, the automatic PR release process is for `next`. That means that releasing to `latest` becomes a manual process. To release latest manually while in "next" mode:
+
+1. *In the code snippets below, replace `0.X` with your version (ex: `0.18`, `release/0.18`, etc.).*
```

Comment:
are there meant to be a double of `1` in the list? otherwise, looks great to me! 

(i think markdown would probably dea with that anyways)

Reply:
yup, markdown automatically handles this which is nice when you end up moving list items around. They should all be `1`s